src/makeInfinityAdder.js

```diff
@@ -5,6 +5,23 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let sum = 0;
+
+  const sumNumber = (number) => {
+    if (number === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += number;
+
+    return sumNumber;
+  };
```

Comment:
The current implementation of `sumNumber` does not allow chaining multiple calls before retrieval of the sum. The function should return a new function when a number is provided, instead of returning itself, to allow the chaining of calls as per the task description.